src/Microsoft.AspNetCore.WebHooks.Common/Services/DefaultAssemblyProvider.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Reflection;
+using Microsoft.Extensions.PlatformAbstractions;
+
+namespace Microsoft.AspNet.WebHooks
+{
+    public class DefaultAssemblyProvider

```

Comment:
BTW starting in RC2 there's a new feature called Application Parts that will make this code much nicer! Will be interesting to see what can be simplified when this is updated to RC2.